src/dev-app/performance/performance-demo.ts

```diff
@@ -0,0 +1,136 @@
+  standalone: true,
+  imports: [
+    CommonModule,
+    FormsModule,
+    MatButtonModule,
+    MatDividerModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatPaginatorModule,
+    MatSelectModule,
+    MatTableModule,
+  ],
+})
+export class PerformanceDemo implements AfterViewInit {
+  /** Controls the rendering of components. */
+  show = false;
+
+  /** The number of times metrics will be gathered. */
+  sampleSize = 100;
+
+  /** The number of components being rendered. */
+  componentCount = 100;
+
+  /** A flat array of every sample recorded. */
+  allSamples: number[] = [];
+
+  /** Used to disable benchmark controls while a benchmark is being run. */
+  isRunningBenchmark = false;
+
+  /** The columns in the metrics table. */
+  displayedColumns: string[] = ['index', 'time'];
+
+  /** Basically the same thing as allSamples but organized as a mat-table data source. */
+  dataSource = new MatTableDataSource<{index: number; time: string}>();
+
+  totalRenderTime = '';
+
+  /** The standard deviation of the recorded samples. */
+  get stdev(): number | undefined {
+    if (!this.allSamples.length) {
+      return undefined;
+    }
+    return Math.sqrt(
+      this.allSamples.map(x => Math.pow(x - this.mean!, 2)).reduce((a, b) => a + b) /
+        this.allSamples.length,
+    );
+  }
+
+  /** The average value of the recorded samples. */
+  get mean(): number | undefined {
+    if (!this.allSamples.length) {
+      return undefined;
+    }
+    return this.allSamples.reduce((a, b) => a + b) / this.allSamples.length;
+  }
+
+  constructor(private _ngZone: NgZone) {}
+
+  @ViewChild(MatPaginator) paginator?: MatPaginator;
+
+  ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator!;
+  }
+
+  getTotalRenderTime(): string {
+    return this.allSamples.length ? `${this.format(this.mean!)} ± ${this.format(this.stdev!)}` : '';
+  }
+
+  format(num: number): string {
+    const roundedNum = Math.round(num * 100) / 100;
+    return roundedNum >= 10 ? roundedNum.toFixed(2) : '0' + roundedNum.toFixed(2);
+  }
+
+  async runBenchmark(): Promise<void> {
+    this.isRunningBenchmark = true;
+    const samples = [];
+    for (let i = 0; i < this.sampleSize; i++) {
+      samples.push(await this.recordSample());
+    }
+    this.dataSource.data = this.dataSource.data.concat(
+      samples.map((sample, i) => ({
+        index: this.allSamples.length + i,
+        time: this.format(sample),
+      })),
+    );
+    this.allSamples.push(...samples);
+    this.isRunningBenchmark = false;
+    this.totalRenderTime = this.getTotalRenderTime();
+  }
+
+  clearMetrics() {
+    this.allSamples = [];
+    this.dataSource.data = [];
+    this.totalRenderTime = this.getTotalRenderTime();
+  }
+
+  recordSample(): Promise<number> {
+    return new Promise(res => {
+      setTimeout(() => {
+        this.show = true;
+        const start = performance.now();
+        this._ngZone.onStable.pipe(take(1)).subscribe(() => {
```

Comment:
It's your call, but as I mentioned before over DM, I don't think that we should rely on zone.js to tell us when to measure, because it can be affected by more than our own code. If we benchmark through the dev tools, it'll be easier to discern what is slowing down a component's rendering.

Reply:
Maybe just add a disclaimer in he UI? e.g. "This is an approximate measure, for more precise performance benchmarking use Chrome devtools"